base/loading.jl

```diff
@@ -230,7 +234,7 @@ This is only needed if your module depends on a file that is not used via `inclu
 no effect outside of compilation.
 """
 function include_dependency(path::AbstractString)
-    _include_dependency(path)
+    _include_dependency("__external__", path)
```

Comment:
Currently `__external__` is a perfectly valid module name, albeit odd. 

Solutions I see:
- make `__external__` a reserved module name.
- add a flag to signify a external file.

Reply:
How about I change it to `#__external__`? That seems pretty much impossible unless someone goes to heroic effort to generate it, `@eval using $(Symbol("#__external__"))`.

Reply:
That sounds good even though I would love a verification step when we write out the module, but maybe that is easier to add to Revise.